packages/story-editor/src/app/rightClickMenu/hooks/useHeadingSelect.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useEffect, useMemo, useState } from '@googleforcreators/react';
+import { __, sprintf } from '@googleforcreators/i18n';
+import { Icons } from '@googleforcreators/design-system';
+import styled from 'styled-components';
+import { getTextElementTagNames } from '@googleforcreators/output';
+
+/**
+ * Internal dependencies
+ */
+import useStory from '../../story/useStory';
+import { useConfig } from '../../config';
+import { RIGHT_CLICK_MENU_LABELS } from '../constants';
+
+const ReversedIcon = styled(Icons.ChevronRightSmall)`
+  transform: rotate(180deg);
+`;
+
+function useHeadingSelect({ menuItemProps, isMenuOpen }) {
+  const { isRTL } = useConfig();
+  const [isSubMenuOpen, setIsSubMenuOpen] = useState(false);
+  const { selectedElements, updateSelectedElements } = useStory(
+    ({ state: { selectedElements }, actions: { updateSelectedElements } }) => ({
+      selectedElements,
+      updateSelectedElements,
+    })
+  );
+
+  useEffect(() => {
+    // Close submenu if the menu itself also closes.
+    if (!isMenuOpen) {
+      setIsSubMenuOpen(false);
+    }
+  }, [isMenuOpen]);
+
+  const subMenuItems = useMemo(() => {
+    if (!isMenuOpen || selectedElements.length === 0) {
+      return [];
+    }
+
+    const HEADING_LEVELS = {
```

Comment:
Nit: we can move this out of the function for consistency, we generally keep the constants defined like this with capital letters out of the functions.

Reply:
Moved

https://github.com/GoogleForCreators/web-stories-wp/pull/12062/commits/aa981e4b0eda565f36f914645c9b03c7788b09ec#diff-a587530b32e93f39df70710cc55d95daffe62546e81293de4d2e64fa441e0f19R36